oak_functions/loader/src/server.rs

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Context;
+use http::{request::Request, response::Response};
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Server, StatusCode,
+};
+use log::info;
+use std::{net::SocketAddr, sync::Arc};
+
+const MAIN_FUNCTION_NAME: &str = "main";
+
+// An ephemeral request handler with a Wasm module for handling the requests.
+#[derive(Clone)]
+pub(crate) struct WasmHandler {
+    // Wasm module to be served on each invocation.
+    module: Arc<wasmi::Module>,
```

Comment:
Not sure I see why this needs to be an `Arc` instead of just `wasmi::Module`. Is it because of the lifetimes involved in `wasmi::ModuleInstance::new`? If so, could you add a comment?